boilerplate/openshift/golang-osd-operator/rvmo-bundle.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -e
+
+# TODO: some syntax error here
+REPOSITORY=${REPOSITORY:"git@github.com:openshift/managed-release-bundle.git"}
+BRANCH=${BRANCH:main}
+
+git clone -b "${BRANCH}" "${REPOSITORY}"
```

Comment:
Do we want to make sure we're in a particular PWD before doing this? Perhaps a safe, fresh `mktemp -d`?

Reply:
> wasn't really thinking about littering

Could be inconvenient/messy for folks testing locally, though?

> I'm assuming that /tmp is writable in the case of the app-interface pipeline?

You would think, wouldn't you? I recently ran into a [thing](https://github.com/openshift/hive/pull/2178) where a tweak to the container config made it not so.

However, boilerplate is definitely doing things like this elsewhere, so you're probably safe.
- [Example](https://github.com/openshift/boilerplate/blob/bbb667c6d0ad14399c2e572acab68dd01ffa3f18/boilerplate/_lib/freeze-check#L53-L62) cloning a repo. (I think this is run in openshift CI, which isn't appsre.) (While you're here, though, note the syntax used to clone just the commit we care about to save space and network traffic.)
- [Example](https://github.com/openshift/boilerplate/blob/94825f3cf075658be2cac702c0b64e6f037ef7c4/boilerplate/openshift/golang-osd-operator/build-opm-catalog.sh#L51) which I'm almost certain _is_ run in appsre (it was even contributed by one of their devs).

As to how you path into that thing, what you're doing ought to work fine from what I see. (Note that the `popd` is redundant unless you're running this script by sourcing it, or adding code after that line.)

Reply:
Updated!

Creating a tmp dir and removing it on exit + added `--single-branch` to the clone (we will be pushing back to the branch with the commited changes, AFAIU we need the entire branch)

Thanks for the feedback!